src/runAll.js

```diff
@@ -24,44 +25,35 @@ module.exports = function runAll(scripts, config) {
   const renderer = config.renderer
   sgf.cwd = resolveGitDir(gitDir)
 
-  return new Promise((resolve, reject) => {
-    sgf('ACM', (err, files) => {
-      if (err) {
-        return reject(err)
-      }
-
-      /* files is an Object{ filename: String, status: String } */
-      const filenames = files.map(file => file.filename)
-      const tasks = generateTasks(config, filenames).map(task => ({
-        title: `Running tasks for ${task.pattern}`,
-        task: () =>
-          new Listr(runScript(task.commands, task.fileList, scripts, config), {
-            // In sub-tasks we don't want to run concurrently
-            // and we want to abort on errors
-            dateFormat: false,
-            concurrent: false,
-            exitOnError: true
-          }),
-        skip: () => {
-          if (task.fileList.length === 0) {
-            return `No staged files match ${task.pattern}`
-          }
-          return false
-        }
-      }))
-
-      if (tasks.length) {
-        return new Listr(tasks, {
+  return pify(sgf)('ACM').then(files => {
+    /* files is an Object{ filename: String, status: String } */
+    const filenames = files.map(file => file.filename)
+    const tasks = generateTasks(config, filenames).map(task => ({
+      title: `Running tasks for ${task.pattern}`,
+      task: () =>
+        new Listr(runScript(task.commands, task.fileList, scripts, config), {
+          // In sub-tasks we don't want to run concurrently
+          // and we want to abort on errors
           dateFormat: false,
-          concurrent,
-          renderer,
-          exitOnError: !concurrent // Wait for all errors when running concurrently
-        })
-          .run()
-          .then(resolve)
-          .catch(reject)
```

Comment:
Also what about this error here?

Reply:
We are chaining promises here. Here's a simple example to show how that works:

```js
'use strict'

function makePromise(fail) {
  return Promise.resolve().then(() => {
    if (fail === true) {
      return Promise.reject('Chained promise rejection')
    }
    return Promise.resolve('Chained promsie resolution')
  })
}

const log = console.log.bind(console)

makePromise(false)
  .then(log)
  .catch(log)

makePromise(true)
  .then(log)
  .catch(log)
```
Output:
```
$ node chain-promise.js
Chained promsie resolution
Chained promise rejection
```